libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -399,6 +401,12 @@ bool AP_OADijkstra::create_inclusion_polygon_with_margin(float margin_cm, AP_OAD
     }
     return true;
 }
+bool AP_OADijkstra::create_inclusion_polygon_with_margin(float margin_cm, AP_OADijkstra_Error &err_id)
+{
+    const bool ret = _create_inclusion_polygon_with_margin(margin_cm, err_id);
+    last_create_inclusion_polygon_with_margin_error_id = err_id;
```

Comment:
couldn't you pass this class variable in from the function that calls this? That is to say remove this:

https://github.com/ArduPilot/ardupilot/blob/bddb9aaac73e777317122057df0ee737e69b599d/libraries/AC_Avoidance/AP_OADijkstra.cpp#L93

And use the class variable in its place. I don't think you would need any of the other changes?